### ISSUE TITLE
Should be able to set global position

### ISSUE THREAD

**Issue by michael-hart** [dronekit/dronekit-python#826]
Particularly for using simulations, instead of setting a home on a map using e.g. Mission Planner, DroneKit should allow the setting of the location of the drone. Currently, the global position is read-only.

**Comment by hamishwillee** [comment]
The simulators allow the setting of initial position themselves on startup. In addition, it doesn't make sense for DroneKit to set a home position that will be overridden immediately by the firmware on arming or whatever.

**Comment by michael-hart** [comment]
Would this still be a useful feature in the case of GPS-denied environments? Other sensor information can provide a location, but it will be local rather than global.

**Comment by hamishwillee** [comment]
I don't think so, because again, in Simulation the vehicle position is from the simulator, and the home position is set by the autopilot. There is a MAVLink command for setting home I believe, but given the actual behaviour of autopilots, it is a lot pointless.

I could of course be wrong, but the discussion would be best held at the autopilot level, and once a case where setting home from an external point makes sense is found, then you could look at implementing it.

**Comment by peterbarker** [comment]
On Fri, 11 May 2018, Hamish Willee wrote:

> 
> I don't think so, because again, in Simulation the vehicle position is from
> the simulator, and the home position is set by the autopilot. There is a
> MAVLink command for setting home I believe, but given the actual behaviour
> of autopilots, it is a lot pointless.

Well, ordinarily in simulations, this is quite true.

But that's because we're ordinarily kind to our vehicles and give them a 
GPS unit.

If you take the GPS unit away, then you have to manually tell the vehicle 
where it is using the mavlink message SET_GPS_GLOBAL_ORIGIN.  This is 
vital for ArduPilot's mavlink-vision-message navigation, for example.

**Comment by michael-hart** [comment]
Is the conclusion of this that either the simulator sets the origin, or it can be set using the SET_GPS_GLOBAL_ORIGIN message?

**Comment by peterbarker** [comment]
On Tue, 15 May 2018, michael-hart wrote:

> Is the conclusion of this that either the simulator sets the origin, or it can be set using the SET_GPS_GLOBAL_ORIGIN message?

The simulated vehicle gets plonked down at the coordinates supplied. 
Ordinarily its simulated onboard GPS would supply location information to 
the ArduPilot code, and the vehicle would eventually become happy it knew 
where it was and set its Origin - and shortly after that its home 
position.

If you do not have simulated sensors on-board to tell you where you are 
then you must tell the vehicle where it is - via mavlink, for example.

Please note Home and Origin are distinct concepts.
